transport/tchannel/handler.go

```diff
@@ -72,18 +70,16 @@ type inboundCallResponse interface {
 	SetApplicationError() error
 }
 
-// responseWriter provides an interface similar to handlerWriter.
-//
-// It allows us to control handlerWriter during testing.
+// responseWriter enhances transport.ResponseWriter interface with transport specific
+// methods.
 type responseWriter interface {
-	AddHeaders(h transport.Headers)
-	AddHeader(key string, value string)
```

Comment:
Renamed to AddSystemHeader to increase visibility and stress difference between "AddHeaders" and "AddHeader": AddSystemHeader doesn't have names validation.